Google.Api.Generator/ProtoUtils/ProtoExtensions.cs

```diff
@@ -78,6 +78,32 @@ internal static string RemoveEnumPrefix(string enumName, string valueName)
             return valueName;
         }
 
+        /// <summary>
+        /// Returns a sequence of messages covering <paramref name="descriptor"/> and all nested messages within it,
+        /// recursively.
+        /// </summary>
+        internal static IEnumerable<MessageDescriptor> SelfAndNestedMessagesRecursively(this MessageDescriptor descriptor)
+        {
+            yield return descriptor;
+            foreach (var nestedType in descriptor.NestedTypes.SelectMany(SelfAndNestedMessagesRecursively))
```

Comment:
Can NestedTypes have enums or just messages. It doesn't matter a lot because the enums won't have resource properties, but it'd be at odds with this method name.

Reply:
NestedTypes is only messages. (There's a separate NestedEnums property.)